rust/value.rs

```diff
@@ -76,57 +143,225 @@ impl Spanned for Literal {
+            TimeZone::ISO(value) => f.write_str(value),
+        }
+    }
+}
+
+impl fmt::Display for DurationDate {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        if let Some(years) = &self.years {
+            write!(f, "{years}Y")?;
+        }
+        if let Some(months) = &self.months {
+            write!(f, "{months}M")?;
+        }
+        if let Some(days) = &self.days {
+            write!(f, "{days}D")?;
+        }
+        Ok(())
+    }
+}
+
+impl fmt::Display for DurationTime {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        if let Some(hours) = &self.hours {
+            write!(f, "{hours}H")?;
+        }
+        if let Some(minutes) = &self.minutes {
+            write!(f, "{minutes}M")?;
+        }
+        if let Some(seconds) = &self.seconds {
+            write!(f, "{seconds}S")?;
+        }
+        Ok(())
+    }
+}
+
+impl fmt::Display for BooleanLiteral {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        f.write_str(self.value.as_str())
+    }
+}
+
+impl fmt::Display for SignedIntegerLiteral {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        if let Some(sign) = &self.sign {
+            fmt::Display::fmt(sign, f)?;
+        }
+        f.write_str(self.integral.as_str())
+    }
+}
+
+impl fmt::Display for SignedDecimalLiteral {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        if let Some(sign) = &self.sign {
+            fmt::Display::fmt(sign, f)?;
+        }
+        f.write_str(self.decimal.as_str())
+    }
+}
+
+impl fmt::Display for DateLiteral {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        fmt::Display::fmt(&self.date, f)
+    }
+}
+
+impl fmt::Display for DateTimeLiteral {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        write!(f, "{}T{}", &self.date, &self.time)
+    }
+}
+
+impl fmt::Display for DateTimeTZLiteral {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        fmt::Display::fmt(&self.date, f)?;
+        fmt::Display::fmt(&self.time, f)?;
+        fmt::Display::fmt(&self.timezone, f)?;
+        Ok(())
+    }
+}
+
+impl fmt::Display for DurationLiteral {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        f.write_str("P")?;
+        match self {
+            DurationLiteral::Weeks(weeks) => {
+                f.write_str(weeks)?;
+                f.write_str("W")?;
+            }
+            DurationLiteral::DateAndTime(date, time) => {
+                fmt::Display::fmt(date, f)?;
+                match time {
+                    None => {}
+                    Some(time) => write!(f, "T{time}")?,
                 }
-                .into());
             }
+        }
+        Ok(())
+    }
+}
+
+impl fmt::Display for StructLiteral {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        f.write_str(self.inner.as_str())
+    }
+}
+
+impl StringLiteral {
+    pub fn parse_to_string(escaped_string: &str) -> Result<String> {
```

Comment:
I've moved this to an impl of StringLiteral, which doesn't feel quite right.

Reply:
This should probably now be called `unescape` or `unquote`